client/state/notification-settings/actions.js

```diff
@@ -97,28 +95,24 @@ export const showSaveErrorNotice = () =>
 export const saveSettings = ( source, settings, applyToAll = false ) => ( dispatch ) => {
 	dispatch( { type: NOTIFICATION_SETTINGS_SAVE } );
 
-	wpcom
-		.undocumented()
-		.me()
-		.updateNotificationSettings(
-			buildSavePayload( source, settings ),
-			applyToAll,
-			( error, data ) => {
-				if ( error ) {
-					dispatch( showSaveErrorNotice() );
-					dispatch( {
-						type: NOTIFICATION_SETTINGS_SAVE_FAILED,
-						error,
-						data,
-					} );
-				} else {
-					dispatch( showSaveSuccessNotice() );
-					dispatch( {
-						type: NOTIFICATION_SETTINGS_SAVE_COMPLETE,
-						error,
-						data,
-					} );
-				}
-			}
-		);
+	const query = applyToAll ? { applyToAll: true } : {};
+
+	wpcom.req
+		.post( '/me/notifications/settings/', query, buildSavePayload( source, settings ) )
+		.then( ( data ) => {
+			dispatch( showSaveSuccessNotice() );
+			dispatch( {
+				type: NOTIFICATION_SETTINGS_SAVE_COMPLETE,
+				error: undefined,
+				data,
+			} );
+		} )
+		.catch( ( error ) => {
+			dispatch( showSaveErrorNotice() );
+			dispatch( {
+				type: NOTIFICATION_SETTINGS_SAVE_FAILED,
+				error,
+				data: undefined,
```

Comment:
This `SAVE_FAILED` action doesn't need the `error` and `data` fields. They are not used. The action is used only to reset an `isFetching` flag.

Similary, `SAVE_COMPLETE` above uses just the `data`, not the `error`.

And neither does `NOTIFICATION_SETTINGS_FETCH_FAILED` use the `error` field.

Reply:
Good call! I'll follow-up in another PR to clean up that reducer.

Reply:
Seems like the cleanup was more minor than I expected, so I could have done it as part of this PR. Anyway, will wrangle a follow-up PR while we're here.

Reply:
#58045 to further clean this up.